lfroms/COMP4109-Project
# Refactor conversations so that the structure is clearer

### What is the problem being solved in this PR?
This PR refactors the concept of conversations so that the structure is clearer, making it easier to implement conversations with 1 or more people.  Note that conversations are immutable: once they are created, users can no longer be added or removed.

### Related issues or PRs



Resolves #37

### How did you accomplish this and why did you choose this approach?
- By renaming "chat" to "conversation" everywhere.
- By moving the `useConversation` hook to `index`.  This loses our ability to subscribe on hotlink, but this will be resolved once we auto-join all conversations after fetching them.
- By renaming the concept of "joining" a conversation to "creating".  Conversations will never be joined.  Instead they will be created and then subscribed to by the server.
- By adjusting the semantics of the payloads.

### How to test changes
1. `yarn start:dev`
2. Go to `localhost:3000` on 2 tabs and join the same conversation. Ensure that messaging works as expected.

### Checklist
